website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * Module that controls the addon node settings. Includes Knockout view-model
+ * for syncing data, and HGrid-folderpicker for selecting a folder.
+ */
+'use strict';
+
+var ko = require('knockout');
+require('knockout.punches');
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+
+
+var ZeroClipboard = require('zeroclipboard');
+ZeroClipboard.config('/static/vendor/bower_components/zeroclipboard/dist/ZeroClipboard.swf');
+var $osf = require('js/osfHelpers');
+var oop = require('js/oop');
+var FolderPickerViewModel = require('js/folderPickerNodeConfig');
+
+ko.punches.enableAll();
+
+/**
+ * View model to support instances of AddonNodeConfig (folder picker widget)
+ *
+ * @class AddonFolderPickerViewModel
+ * @param {String} addonName Full display name of the addon
+ * @param {String} url API url to initially fetch settings
+ * @param {String} selector CSS selector for containing div
+ * @param {String} folderPicker CSS selector for folderPicker div
+ * @param {Object} opts Optional overrides to the class' default treebeardOptions, in particular onPickFolder
+ */
+var OauthAddonFolderPickerViewModel = oop.extend(FolderPickerViewModel, {
+    constructor: function(addonName, url, selector, folderPicker, opts) {
+        var self = this;
+        self.super.constructor.call(self, addonName, url, selector, folderPicker);
+        // whether the auth token is valid
+        self.validCredentials = ko.observable(true);
```

Comment:
`self.validCredentials` and `self.loading` are already in the FolderPickerViewModel constructor so I think those can be removed.